x/emissions/keeper/msgserver/msg_server_requests.go

```diff
@@ -119,9 +119,25 @@ func (ms msgServer) RequestInference(ctx context.Context, msg *types.MsgRequestI
 	}
 	// 12. Write request state into the mempool state
 	request.BlockLastChecked = currentBlock
-	err = ms.k.AddToMempool(ctx, *request)
+	unmetDemand, err := ms.k.AddToMempool(ctx, *request)
 	if err != nil {
 		return nil, err
 	}
+	// 13. Activate topic if meet demand
+	isActivated, err := ms.k.IsTopicActive(ctx, request.TopicId)
+	if err != nil {
+		return nil, err
+	}
+	if !isActivated {
+		minTopicUnmentDemand, err := ms.k.GetParamsMinTopicUnmetDemand(ctx)
+		if err != nil {
+			return nil, err
+		}
+		minTopicUnmetDemandUint := cosmosMath.NewUintFromString(minTopicUnmentDemand.String())
+
+		if unmetDemand.GTE(minTopicUnmetDemandUint) {
+			_ = ms.k.ActivateTopic(ctx, request.TopicId)
```

Comment:
check error here

Reply:
Do you think we should revert this request when activate topic failed?

Reply:
yeah, I mean, it should work properly. The user can be expecting that his request will activate the topic, it can be the main goal of the tx. The probability of getting an error at this point is near zero but it doesn't hurt to check.

Reply:
I mean If it generates an error at this point, it probably something wrong in protocol, so it's also good to have it for testing purposes.